app/models/provisioning_template.rb

```diff
@@ -44,28 +35,48 @@ class ConfigTemplate < ActiveRecord::Base
   scoped_search :in => :hostgroups,       :on => :name, :rename => :hostgroup,       :complete_value => true
   scoped_search :in => :template_kind,    :on => :name, :rename => :kind,            :complete_value => true
 
-  class Jail < Safemode::Jail
-    allow :name
+  # Override method in Taxonomix as Template is not used attached to a Host,
+  # and matching a Host does not prevent removing a template from its taxonomy.
+  def used_taxonomy_ids(type)
+    []
   end
 
-  def clone
-    self.deep_clone(:include => [:operatingsystems, :organizations, :locations],
-                    :except  => [:name, :locked, :default, :vendor])
-  end
+  # with proc support, default_scope can no longer be chained
+  # include all default scoping here
+  default_scope lambda {
+    with_taxonomy_scope do
+      order("#{Template.table_name}.name")
+    end
+  }
 
   # TODO: review if we can improve SQL
   def self.template_ids_for(hosts)
     hosts.with_os.map do |host|
-      host.configTemplate.try(:id)
+      host.provisioning_template.try(:id)
     end.uniq.compact
   end
 
+  # not sure why but this changes table_name so we set it explicitly (same does not apply for Ptable)
+  def self.base_class
+    self
```

Comment:
basically this
http://apidock.com/rails/ActiveRecord/Associations/ClassMethods/has_many#1010-Polymorphic-has-many-within-inherited-class-gotcha

not sure if taxonomies or audits or both saved type as `Template` instead of child class